earthpy/mask.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+import numpy.ma as ma
+
+
+def make_cloud_mask(mask_arr, vals):
+    """Take an input single band mask layer such as a pixel_qa
+    layer for MODIS or Landsat and apply a mask given a range of vals to mask
+
+    Parameters
+    -----------
+    mask_arr : numpy array
+        A array... to open the pixel_qa or mask raster of interest
+    vals : list of numbers (int or float)
+        A list of values that represent no data in the provided raster
+        layer (arr)
+
+    Returns
+    -----------
+    arr : numpy array
+        A numpy array with values that should be masked set to 1 for
+        True (Boolean)
+    """
+
+    # Improved by Joe! Not tested yet but should work!
+    temp_mask = np.isin(mask_arr, vals)
+    mask_arr[temp_mask] = 1
+    # for cval in vals:
+    #    mask_arr[mask_arr == cval] = 1
+    return(mask_arr)
+
+
+def apply_cloud_mask(arr, the_mask):
+    # Create a mask for all bands in the landsat scene
+    cl_mask = np.broadcast_to(the_mask == 1,
+                              arr.shape)
+
+    # If the user provides a masked array, combine masks
+    if isinstance(arr, np.ma.MaskedArray):
+        cl_mask = np.logical_or(arr.mask, cl_mask)
+
+    # Return combined mask
+    return (ma.masked_array(arr, mask=cl_mask))
+
+
+def make_apply_mask(arr, mask_arr, vals):
+    cl_mask = make_cloud_mask(mask_arr, vals, arr)
```

Comment:
well this is what i wanted to chat with you about @joemcglinchy . So i can see a few different use cases and i'm not sure which one to use but i decided to go modular. Here's why

## Use Case One
You have 9 landsat tif files - each one needs to be masked and for whatever reason you are processing them individually - without stacking. In that scenario, you want to 
1. Create the mask ONCE as it's used for each tif and then
2. apply the mask to each and every layer. 

You don't want to create the mask over and over.

## Use Case Two
In this use case you have one landsat scene that you open and turn into a stacked numpy array. In that scenario you want to just make and apply the mask all in one shot!

I wrote the functions to allow for each use case. But i do tend to over think and complicate things so please let me know what you think about htis process.

this relates to the stack function as well. Why? Because right now the stack function writes out a .tif file by default which i really dislike. if we were always stacking multi-band rasters before masking, then i think id go with what you suggest above!

Let me know what you think. Originally that stack function was an attempt to align with rasterio needs. I could check back in with them on whether it's something they might want to own. OR we could just decide to meet our needs which i think is to have a stack function that just returns

1. a numpy array and 
2. .meta 

thoughts?